com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -229,11 +229,19 @@ public static ObservationProto GetObservationProto(this ISensor sensor, WriteAda
             }
             else
             {
+                var compressedObs = sensor.GetCompressedObservation();
                 observationProto = new ObservationProto
                 {
-                    CompressedData = ByteString.CopyFrom(sensor.GetCompressedObservation()),
                     CompressionType = (CompressionTypeProto)sensor.GetCompressionType(),
                 };
+
+                // Passing a null byte[] to CompressedData will raise an exception, so leave it unset
+                // if the observations are null.
+                if (compressedObs != null)
```

Comment:
will null data be handled ok on the python side?

Reply:
Probably not. I'll change this back to raise with a clearer error.